hilti/toolchain/src/ast/operators/map.cc

```diff
@@ -255,7 +255,7 @@ class IndexConst : public Operator {
     }
 
     QualifiedTypePtr result(Builder* builder, const Expressions& operands, const Meta& meta) const final {
-        return operands[0]->type()->type()->as<type::Map>()->valueType();
+        return operands[0]->type()->type()->as<type::Map>()->valueType()->recreateAsConst(builder->context());
```

Comment:
Nit: I wonder whether we need similar treatment for iterators into const collections which we do not seem to model.

Reply:
I'll check.